src/test/java/org/broadinstitute/hellbender/tools/copynumber/CreateBinningIntervalsIntegrationTest.java

```diff
@@ -0,0 +1,221 @@
+        binsExpected.add(new Interval(contig, 10001, 20000));
+        binsExpected.add(new Interval(contig, 20001, 20500));
+
+        // compare the file contents to the expected result
+        Assert.assertEquals(binsFromFile, binsExpected);
+    }
+
+
+    // Test for intervals reaching the ends of the reads
+    @Test
+    public void simpleIntervalTest3() {
+        final int widthOfBins = 10000;
+        final int padding = 500;
+        final File outputFile = createTempFile("GATK-create-binning-intervals-test", ".tmp");
+        final String[] args = {
+                "-L", "20:3000-20000",
+                "-L", "20:63025220-63025520",
+                "-R", REFERENCE.getAbsolutePath(),
+                "-" + CreateBinningIntervals.WIDTH_OF_BINS_SHORT_NAME, Integer.toString(widthOfBins),
+                "-" + CreateBinningIntervals.PADDING_SHORT_NAME, Integer.toString(padding),
+                "-O", outputFile.getAbsolutePath()
+        };
+        runCommandLine(args);
+
+        // read the contents of the output file and print them out
+        // printOutputFile(outputFile);
+
+        // read the bins from the output file
+        IntervalList binsFromFile = IntervalList.fromFile(outputFile);
+
+        // we expect the following result
+        IntervalList binsExpected = new IntervalList(binsFromFile.getHeader().getSequenceDictionary());
+        String contig = binsFromFile.iterator().next().getContig();
+        binsExpected.add(new Interval(contig, 2500, 12499));
+        binsExpected.add(new Interval(contig, 12500, 20500));
+        binsExpected.add(new Interval(contig, 63024720, 63025520));
+
+        // compare the file contents to the expected result
+        Assert.assertEquals(binsFromFile, binsExpected);
+    }
+
+    // Test for whole genome
+    @Test
+    public void simpleIntervalTest4() {
+        final int widthOfBins = 10_000_000;
+        final int padding = 500;
+        final File outputFile = createTempFile("GATK-create-binning-intervals-test", ".tmp");
+        final String[] args = {
+                "-L", "20:1-63025520",
+                "-R", REFERENCE.getAbsolutePath(),
+                "-" + CreateBinningIntervals.WIDTH_OF_BINS_SHORT_NAME, Integer.toString(widthOfBins),
+                "-" + CreateBinningIntervals.PADDING_SHORT_NAME, Integer.toString(padding),
+                "-O", outputFile.getAbsolutePath()
+        };
+        runCommandLine(args);
+
+        // read the contents of the output file and print them out
+        // printOutputFile(outputFile);
+
+        // read the bins from the output file
+        IntervalList binsFromFile = IntervalList.fromFile(outputFile);
+
+        // we expect the following result
+        IntervalList binsExpected = new IntervalList(binsFromFile.getHeader().getSequenceDictionary());
+        String contig = binsFromFile.iterator().next().getContig();
+        binsExpected.add(new Interval(contig, 1, 10000000));
```

Comment:
I think how you have it is fine, but if this chromosome were a bit longer you could render this concise with:
```java
final int width = 10_000_000;
IntStream.rangeClosed(0, 6)
   .mapToObj(n -> new Interval(contig, n * width + 1, Math.min( (n + 1) * width, contigSize)))
   .forEach(binsExpected::add);
```

Reply:
Thanks, I will do that next time, if I have more intervals.